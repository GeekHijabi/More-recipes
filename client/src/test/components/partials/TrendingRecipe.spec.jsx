import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedTrendingRecipe, { TrendingRecipe }
  from '../../../js/components/Partials/TrendingRecipes';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiGetRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  recipes: []
};
describe('Component: TrendingRecipe', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });

  describe('TrendingRecipe in snapshot', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<TrendingRecipe {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Connected TrendingRecipe component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          favRecipes: []
        }
      });
      const wrapper = shallow(<ConnectedTrendingRecipe store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});

