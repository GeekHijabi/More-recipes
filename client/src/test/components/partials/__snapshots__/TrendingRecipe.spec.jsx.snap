// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TrendingRecipe TrendingRecipe in snapshot should render correctly 1`] = `
<div
  style={
    Object {
      "margin": "0 6.5em",
      "width": "85%",
    }
  }
>
  <h4
    className="heading"
  >
    <span>
      Currently Trending
    </span>
  </h4>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
